### ISSUE TITLE
Semi-transparent colors do not display properly in PGraphics

### ISSUE THREAD

**Issue by cadin** [processing/processing#5519]
## Description

Drawing to PGraphics with non-opaque alpha causes the color/alpha to be drawn incorrectly.
It appears that the color is drawn at half of the specified alpha.

## Expected Behavior

The color should be drawn with the specified values, composited against the graphics that appear underneath the image. This works properly in Processing 2.

## Current Behavior

In Processing 3, the color appears by default to be composited against black (the color appears muddy-grey). It also appears to be drawn at a lower alpha value than what is specified in the `fill()` call.

![pgraphicsexample](https://user-images.githubusercontent.com/2348891/40637427-a306c382-62b9-11e8-87f0-da996ff8c9e2.png)

## Steps to Reproduce

Reproduce with this simple sketch:

	size(200, 200);
	background(255);

	color red = color(255, 0, 0);
	PGraphics pg = createGraphics(100, 200);
	pg.beginDraw();
	// pg.background(255, 0)
	pg.noStroke();
	pg.fill(red, 128);
	pg.rect(0,0,100,200);
	pg.endDraw();
	image(pg, 0, 0);

Note, calling `background()` with transparent white fixes the gray issue, but the transparency is still incorrect.

## Environment

 

* Processing version: 3.3.7
* Operating System and OS version: macOS 10.13.4
* Other information: MacBook Pro with NVIDIA GeForce GT 750M 2048 MB

**Comment by cadin** [comment]
Here's what I get on my system:
```
bg: FF00FF00

1: FFFF0000
2: C080007F
3: C040007F
4: 800000FF

1: FFFF0000
2: FF603F60
3: FF303F60
4: FF007F80

[default]
cm: DirectColorModel: rmask=ff0000 gmask=ff00 bmask=ff amask=0
premult: false

[opaque]
cm: DirectColorModel: rmask=ff0000 gmask=ff00 bmask=ff amask=0
premult: false

[translucent]
cm: DirectColorModel: rmask=ff0000 gmask=ff00 bmask=ff amask=ff000000
premult: true

premult g: true
premult pg: true

```

**Comment by JakubValtar** [comment]
Thanks a lot. For some reason we are getting premultiplied contexts on your machine (I'm gonna assume it happens on all macOS machines). I will submit a fix.

For the record, it works well on my end. These are the values I get on Windows 10 (1803):
```
bg: FF00FF00

1: FFFF0000
2: C0AA0055
3: C0550055
4: 800000FF

1: FFFF0000
2: FF803F40
3: FF403F40
4: FF007F80

[default]
cm: DirectColorModel: rmask=ff0000 gmask=ff00 bmask=ff amask=0
premult: false

[opaque]
cm: DirectColorModel: rmask=ff0000 gmask=ff00 bmask=ff amask=0
premult: false

[translucent]
cm: DirectColorModel: rmask=ff0000 gmask=ff00 bmask=ff amask=ff000000
premult: false

premult g: false
premult pg: false

```

**Comment by liquid600pgm** [comment]
I can confirm that this also happened to me on Windows, read [this post on the Processing forum](forum.processing.org/two/discussion/27841/pgraphics-text-has-ugly-edges). The ugly edges of the text seem to be caused by the same problem.

**Comment by JakubValtar** [comment]
Thanks, the fix is ready and will be included in the next release (3.3.8 or 3.4, whichever comes out first).

**Comment by mattkanenet** [comment]
Please note that this remains broken for the P2D and P3D renderers. The same behavior occurs that Caden originally mentioned, but in the case where P2D or P3D renderer is applied to the PGraphics object. This was the use-case fix I was looking for, but took for granted that it would be renderer specfic. 

`size(200, 200, P2D);
background(255);

color red = color(255, 0, 0);
PGraphics pg = createGraphics(100, 200, P2D);
pg.beginDraw();
 //pg.background(255, 0);
pg.noStroke();
pg.fill(red, 128);
pg.rect(0,0,100,200);
pg.endDraw();
image(pg, 0, 0);`

Environment
Processing version: 3.4
Operating System and OS version: Windows 7   64bit
Other information: HP 810-145qe   Intel Core i7-4790 CPU @ 3.6GHZ
Video Card: NVidia GeForce GTX 745
